clang/lib/Sema/SemaBounds.cpp

```diff
@@ -1453,23 +1453,78 @@ namespace {
 
         // Gather free variables.
         for (const auto &SrcV : SrcVars) {
-          if (IsEqualToConstant(S, SrcV, EquivExprs))
+          DeclRefExpr *SrcVar = cast<DeclRefExpr>(SrcV);
```

Comment:
What happens if `SrcV` is not a `DeclRefExpr`?

Reply:
The list `SrcVars` should always be a list of `DeclRefExprs` since it is obtained by searching for all the `DeclRefExprs` that occur in a bounds expression. As a separate cleanup item, the type of `SrcVars` should be changed from `EqualExprTy` (set of `Exprs`) to a set of `DeclRefExprs`.

Reply:
Created #941 to track this.